src/unix/bsd/freebsdlike/dragonfly/mod.rs

```diff
@@ -193,6 +195,10 @@ s! {
     pub struct cpumask_t {
         ary: [u64; 4],
     }
+
+    pub struct __c_anonymous_pthread_spinlock_s {
```

Comment:
it isn't exactly my area but I think if the OS define a struct as opaque, it should be kept opaque.

`pthread_spinlock_t` is opaque in DragonflyBSD too: [sys/sys/_pthreadtypes.h](https://github.com/DragonFlyBSD/DragonFlyBSD/blob/c163a4d7ee9c6857ee4e04a3a2cbb50c3de29da1/sys/sys/_pthreadtypes.h)